google/resource_compute_instance.go

```diff
@@ -1077,6 +1147,33 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 		d.SetPartial("scheduling")
 	}
 
+	if d.HasChange("desired_status") {
+		desiredStatus := d.Get("desired_status").(string)
```

Comment:
Should I check for `d.Get("desired_status") == nil`? If I understand correctly, the `if d.HasChange("desired_status")` part ensure that `d.Get("desired_status")` is not `nil`, but I would like to be sure :)

Reply:
See https://github.com/terraform-providers/terraform-provider-google/pull/4797#discussion_r377787758 (same rules apply)